### DISCUSSION TITLE
kernel command line set static ip only use mac addres

### DISCUSSION THREAD

**Discussion by IAseven** [ipxe/ipxe#1213] [Q&A] [answered]
I know this is not an issue with IPxe, but I have searched many articles but have not found a solution. I would like to consult experts here to see if there are any good solutions,

Here is my question:

I am booting the installation program of Centos7 through IPxe to install the machine, and I need to set a static IP in the kernel command line. These are my parameters:
`Ip=<client ip>:<srv ip>:<gw ip>:<netmask>:<host>:<device>:<autoconf>`

`<device>`is the corresponding network card name, but I am not sure what that name is, I only know that the MAC address of the network card needs to be set,

May I ask how to set a static IP based on the MAC when I don't know the name of the device in this situation

Thank you very much to the experts for giving me some feedback.

**Comment by NiKiZe** [comment]
You can't, if there is one nic then eth0, if multiple NICs is present and loaded as modules, then in most distros it not be known.

The ip cmdline is from a time when you built your own kernel for network boot, it is likely that any bootup script would still configure network, and drop your IP config.

**Comment by mcb30** [comment, marked as answer]
As far as I am aware, CentOS uses the Anaconda installer, so you should be able to find the relevant kernel command-line documentation at https://anaconda-installer.readthedocs.io/en/latest/boot-options.html

**Reply by IAseven** [reply]
Thanks for your reply, I found this option,

Anaconda : 
```
ifname
ifname=<interface>:<MAC>
Assign the given interface name to the network device with the given MAC. May be used multiple times.
```

my ipxe script:
```
:install
echo download-kernel...sleep-3s && sleep 3 \
&& kernel ${isoUrl}/images/pxeboot/vmlinuz \
inst.repo=${isoUrl}/ \
ifname=net0:08:00:27:18:7a:eb \
ip=10.0.2.15::10.0.2.2:255.255.255.0::net0:off \
&& initrd ${isoUrl}/images/pxeboot/initrd.img || shell
```

Note:  !!!!!!!!!
ifname=net0:mac 
Do not set the network card name to eth0, if you set it to eth0, the network will not start!!!!
